static/js/plotjs/stackf.js

```diff
@@ -0,0 +1,115 @@
+/* eslint-disable */
+const binsize = slider.value;
+const fluxalph = ((binsize === 0) ? 1.0 : 0.1);
+
+for (let i = 0; i < toggle.labels.length; i++) {
+  const fluxsource = eval(`obs${i}`).data_source;
+  const binsource = eval(`bin${i}`).data_source;
+
+  const fluxerrsource = eval(`obserr${i}`).data_source;
+  const binerrsource = eval(`binerr${i}`).data_source;
+
+  const minmjd = Math.min.apply(Math, fluxsource.data.mjd) - 15;
+  const maxmjd = Math.max.apply(Math, fluxsource.data.mjd) + 15;
+
+  binsource.data.mjd = [];
+  binsource.data.flux = [];
+  binsource.data.fluxerr = [];
+  binsource.data.filter = [];
+  binsource.data.color = [];
+  binsource.data.lim_mag = [];
+  binsource.data.mag = [];
+  binsource.data.magerr = [];
+  binsource.data.instrument = [];
+  binsource.data.stacked = [];
+
+  binerrsource.data.xs = [];
+  binerrsource.data.ys = [];
+  binerrsource.data.color = [];
+
+  for (let j = 0; j < fluxsource.get_length(); j++) {
+      fluxsource.data.alpha[j] = fluxalph;
+      fluxerrsource.data.alpha[j] = fluxalph;
+  }
+
+  if (binsize > 0) {
+    // now do the binning
+    const k = 0;
+    let curmjd = minmjd;
+    const mjdbins = [curmjd];
+
+    while (curmjd < maxmjd) {
+      curmjd += binsize;
+      mjdbins.push(curmjd);
+    }
+
+    const nbins = mjdbins.length - 1;
+    for (let l = 0; l < nbins; l++) {
+      // calculate the flux, fluxerror, and mjd of the bin
+      const flux = [];
+      const weight = [];
+      const mjd = [];
+      const limmag = [];
+      let ivarsum = 0;
+
+      for (let m = 0; m < fluxsource.get_length(); m++) {
+        if ((fluxsource.data.mjd[m] < mjdbins[l + 1]) && (fluxsource.data.mjd[m] >= mjdbins[l])) {
+          const fluxvar = fluxsource.data.fluxerr[m] * fluxsource.data.fluxerr[m];
+          const ivar = 1 / fluxvar;
+
+          weight.push(ivar);
+          flux.push(fluxsource.data.flux[m]);
+          mjd.push(fluxsource.data.mjd[m]);
+          limmag.push(fluxsource.data.lim_mag);
+          ivarsum += ivar;
+        }
+      }
+
+      let myflux = 0;
+      let mymjd = 0;
+
+      if (weight.length === 0) {
+        continue;
+      }
+
+      for (let n = 0; n < weight.length; n++) {
+        myflux += weight[n] * flux[n] / ivarsum;
+        mymjd += weight[n] * mjd[n] / ivarsum;
+      }
+
+      const myfluxerr = Math.sqrt(1 / ivarsum);
+
+
+      if (myflux / myfluxerr > 5.0) {
```

Comment:
Do you want to hardcode this threshold or make it changeable to the end user on the front end (or a config file)?